.scripts/IdleTimeout.py

```diff
@@ -0,0 +1,55 @@
+#Note this script requires super user privilages if the the user is not in the "input" group
+import struct
+from time import sleep
+from datetime import datetime, timedelta
+from os import set_blocking, listdir, path
+from subprocess import call, Popen
+
+pollTime = 10 #Seconds
+timeout = 10  #minutes
+processName = 'emulationstation'
+lastInput = datetime.now()
+inputs = []
+
+def has_input(file): 
+    data = file.read(24)
+    if data != None: #Event file may be empty if no input has been registered for some time
+        return True
+    else:
+        return False
+
+def get_last_input(file):
+    try: #Just in case the file is reset in between calling has_input() and here
+        data = file.read()[:-24]#Read LAST input
```

Comment:
Could we reuse the data which is read in `has_input`, rather then reading the file twice?

This would also potentially fix the problem of the file changing since the last read.

Reply:
True. I've simplified this to just use has_input(). Because we only care if there has been _any_ input the script just tests whether the event file is non-empty. If so it sets the last input to `datetime.now()`. This skews when the last input actually was by a couple of seconds, but shouldn't have a meaningful impact taking into account the time scales we're dealing with.

```
while True:
    for f in inputs:
        if(has_input(f)):
            lastInput = datetime.now()
```